nesting.lua

```diff
@@ -51,6 +51,23 @@ function nesting.resizeNestedAs(output, input)
    end
 end
 
+-- Copies all tensors in the output.
+function nesting.copyNested(output, input)
+   if torch.isTensor(output) then
+      output:copy(input)
+   else
+      for key, child in pairs(input) do
+          nesting.copyNested(output[key], child)
+      end
+      -- Extra elements are removed from the output.
+      for key, child in pairs(output) do
```

Comment:
The removing of extra elements should not be needed. Or an error should be raised if the sizes do not match.